src/Adapter/SearchEngine/CommandHandler/EditSearchEngineHandler.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\SearchEngine\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\SearchEngine\AbstractSearchEngineHandler;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\Command\EditSearchEngineCommand;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\CommandHandler\EditSearchEngineHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\Exception\SearchEngineException;
+use PrestaShopException;
+
+/**
+ * Handles command what edits search engine.
+ */
+final class EditSearchEngineHandler extends AbstractSearchEngineHandler implements EditSearchEngineHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws SearchEngineException
+     */
+    public function handle(EditSearchEngineCommand $command): void
+    {
+        $searchEngine = $this->getSearchEngine($command->getSearchEngineId());
+
+        if (null !== $command->getServer()) {
+            $searchEngine->server = $command->getServer();
+        }
+
+        if (null !== $command->getGetVariable()) {
+            $searchEngine->getvar = $command->getGetVariable();
+        }
+
+        try {
+            if (false === $searchEngine->validateFields(false)) {
+                throw new SearchEngineException('Search engine contain invalid field values.');
```

Comment:
Is there a reason not to have specific error codes for these error cases ? (like you did for other cases)

Reply:
All invalid field errors should get caught by the Symfony form constraints. If fields are invalid there, then something is very wrong.